### ISSUE TITLE
[Crash] Some playlists, potential race condition.

### ISSUE THREAD

**Issue by PeterHindes** [TeamNewPipe/NewPipe#3218]
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I checked if the issue/feature exists in the latest version.
- [x] I did use the [incredible bugreport to markdown converter](https://teamnewpipe.github.io/CrashReportToMarkdown/) to paste bug reports.

[Video Demo](https://imgur.com/a/es8rW0j)

## Exception
* __User Action:__ requested playlist
* __Request:__ https://www.youtube.com/playlist?list=OLAK5uy_nanAsLd8bl9PNfMNXeijLKPxF1kwrCHXA
* __Content Language:__ US
* __Service:__ YouTube
* __Version:__ 0.18.6
* __OS:__ Linux Android 10 - 29
Crashes with un-cached playlists most of the time.

<details><summary><b>Crash log</b></summary><p>

```
java.lang.NullPointerException: Attempt to invoke virtual method 'com.grack.nanojson.JsonObject com.grack.nanojson.JsonObject.getObject(java.lang.String)' on a null object reference
	at org.schabi.newpipe.extractor.services.youtube.linkHandler.YoutubeParsingHelper.defaultAlertsCheck(YoutubeParsingHelper.java:399)
	at org.schabi.newpipe.extractor.services.youtube.extractors.YoutubePlaylistExtractor.onFetchPage(YoutubePlaylistExtractor.java:43)
	at org.schabi.newpipe.extractor.Extractor.fetchPage(Extractor.java:56)
	at org.schabi.newpipe.extractor.playlist.PlaylistInfo.getInfo(PlaylistInfo.java:29)
	at org.schabi.newpipe.util.ExtractorHelper.lambda$getPlaylistInfo$8(ExtractorHelper.java:155)
	at org.schabi.newpipe.util.-$$Lambda$ExtractorHelper$oXX6037sP0Rn784EpEbKFh6xH48.call(Unknown Source:4)
	at io.reactivex.internal.operators.single.SingleFromCallable.subscribeActual(SingleFromCallable.java:44)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.single.SingleDoOnSuccess.subscribeActual(SingleDoOnSuccess.java:35)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.maybe.MaybeFromSingle.subscribeActual(MaybeFromSingle.java:41)
	at io.reactivex.Maybe.subscribe(Maybe.java:4154)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.drain(MaybeConcatArray.java:153)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray$ConcatMaybeObserver.request(MaybeConcatArray.java:78)
	at io.reactivex.internal.operators.flowable.FlowableElementAtMaybe$ElementAtSubscriber.onSubscribe(FlowableElementAtMaybe.java:66)
	at io.reactivex.internal.operators.maybe.MaybeConcatArray.subscribeActual(MaybeConcatArray.java:42)
	at io.reactivex.Flowable.subscribe(Flowable.java:14479)
	at io.reactivex.internal.operators.flowable.FlowableElementAtMaybe.subscribeActual(FlowableElementAtMaybe.java:36)
	at io.reactivex.Maybe.subscribe(Maybe.java:4154)
	at io.reactivex.internal.operators.maybe.MaybeToSingle.subscribeActual(MaybeToSingle.java:46)
	at io.reactivex.Single.subscribe(Single.java:3438)
	at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run(SingleSubscribeOn.java:89)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)

```
</p></details>
<hr>

**Comment by PeterHindes** [comment]
This may be limited to youtube-music playlists (albums auto generated). More testing is needed

**Comment by PeterHindes** [comment]
`{alerts=[{alertWithActionsRenderer={alertMessage={runs=[{text=Get high-quality audio, gapless playback and personalised music recommendations.}]}, trackingParams=CDoQuKIBGAAiEwi--dz61pXoAhV8QEwIHUaqAMc=, icon={iconType=YOUTUBE_MUSIC}, actionButtons=[{buttonRenderer={trackingParams=CDsQjbsFGAAiEwi--dz61pXoAhV8QEwIHUaqAMc=, icon={iconType=EXTERNAL_LINK}, iconPosition=BUTTON_ICON_POSITION_TYPE_RIGHT_OF_TEXT, style=STYLE_PRIMARY, text={runs=[{text=Enjoy now}]}, navigationEndpoint={commandMetadata={webCommandMetadata={rootVe=83769, url=https://music.youtube.com/playlist?list=OLAK5uy_m82c10WquK6-sK_y_gSEOvdx6obxHQd5s}}, clickTrackingParams=CDsQjbsFGAAiEwi--dz61pXoAhV8QEwIHUaqAMc=, urlEndpoint={url=https://music.youtube.com/playlist?list=OLAK5uy_m82c10WquK6-sK_y_gSEOvdx6obxHQd5s, target=TARGET_NEW_WINDOW}}}}], isVisible=true, title={runs=[{text=Try YouTube Music}]}}}], trackingParams=CAAQhGciEwi--dz61pXoAhV8QEwIHUaqAMc=, contents={twoColumnBrowseResultsRenderer={tabs=[{tabRenderer={trackingParams=CDwQ8JMBGAEiEwi--dz61pXoAhV8QEwIHUaqAMc=, selected=true, content={sectionListRenderer={trackingParams=CD0Qui8iEwi--dz61pXoAhV8QEwIHUaqAMc=, contents=[{itemSectionRenderer={trackingParams=CD4Quy8YACITCL753PrWlegCFXxATAgdRqoAxw==, contents=[{playlistVideoListRenderer={playlistId=OLAK5uy_m82c10WquK6-sK_y_gSEOvdx6obxHQd5s, trackingParams=CD8Q7zsYACITCL753PrWlegCFXxATAgdRqoAxw==, onReorderEndpoint={commandMetadata={webCommandMetadata={sendPost=true, apiUrl=/youtubei/v1/browse/edit_playlist, url=/service_ajax}}, playlistEditEndpoint={playlistId=OLAK5uy_m82c10WquK6-sK_y_gSEOvdx6obxHQd5s, params=CAE%3D, actions=[{action=ACTION_MOVE_VIDEO_AFTER}]}, clickTrackingParams=CD8Q7zsYACITCL753PrWlegCFXxATAgdRqoAxw==}, isEditable=false, contents=[{playlistVideoRenderer={lengthText={simpleText=3:25, accessibility={accessibilityData={label=3 minutes, 25 seconds}}}, isPlayable=true, thumbnail={webThumbnailDetailsExtensionData={isPreloaded=true}, thumbnails=[{width=490, url=https://i.ytimg.com/vi/QdabIfmcqSQ/hqdefault.jpg?sqp=-oaymwEZCNACELwBSFXyq4qpAwsIARUAAIhCGAFwAQ==&rs=AOn4CLDozrkHb8QrXdA5vhhEvBTTNs-khQ, height=274}]}, index={simpleText=1}, lengthSeconds=205, videoId=QdabIfmcqSQ, isWatched=false, title={simpleText=Caravan Palace - Miracle (official video), accessibility={accessibilityData={label=Caravan Palace - Miracle (official video) by CaravanPalace 1 year ago 3 minutes, 25 seconds 7,311,600 views}}}, shortBylineText={runs=[{text=CaravanPalace, navigationEndpoint={commandMetadata={webCommandMetadata={rootVe=3611, webPageType=WEB_PAGE_TYPE_BROWSE, url=/channel/UCKH9HfYY_GEcyltl2mbD5lA}}, clickTrackingParams=CFQQxjQYACITCL753PrWlegCFXxATAgdRqoAxw==, browseEndpoint={browseId=UCKH9HfYY_GEcyltl2mbD5lA, canonicalBaseUrl=/channel/UCKH9HfYY_GEcyltl2mbD5lA}}}]}, menu={menuRenderer={trackingParams=CFQQxjQYACITCL753PrWlegCFXxATAgdRqoAxw==, accessibility={accessibilityData={label=Action menu}}, items=[{menuServiceItemRenderer={trackingParams=CFUQ_pgEGAIiEwi--dz61pXoAhV8QEwIHUaqAMc=, icon={iconType=ADD_TO_QUEUE_TAIL}, text={runs=[{text=Add to queue}]}, serviceEndpoint={commandMetadata={webCommandMetadata={sendPost=true, url=/service_ajax}}, clickTrackingParams=CFUQ_pgEGAIiEwi--dz61pXoAhV8QEwIHUaqAMc=, signalServiceEndpoint={signal=CLIENT_SIGNAL, actions=[{addToPlaylistCommand={onCreateListCommand={commandMetadata={webCommandMetadata={sendPost=true, apiUrl=/youtubei/v1/playlist/create, url=/service_ajax}}, createPlaylistServiceEndpoint={hack=true, params=CAQ%3D, videoIds=[QdabIfmcqSQ]}, clickTrackingParams=CFUQ_pgEGAIiEwi--dz61pXoAhV8QEwIHUaqAMc=}, videoId=QdabIfmcqSQ, openMiniplayer=true, openListPanel=false, listType=PLAYLIST_EDIT_LIST_TYPE_QUEUE, videoIds=[QdabIfmcqSQ]}}]}}}}]}}, thumbnailOverlays=[{thumbnailOverlayTimeStatusRenderer={style=DEFAULT, text={simpleText=3:25, accessibility={accessibilityData={label=3 minutes, 25 seconds}}}}}, {thumbnailOverlayNowPlayingRenderer={text={runs=[{text=Now playing}]}}}], trackingParams=CFQQxjQYACITCL753PrWlegCFXxATAgdRqoAxw==, navigationEndpoint={commandMetadata={we`

Here is some of the data that was passed that caused the error.

**Comment by Stypox** [comment]
Closed by #323
